### DISCUSSION TITLE
Metric Aggregation Sum not working - Need help

### DISCUSSION THREAD

**Discussion by the-arcade-01** [jertel/elastalert2#1138] [Q&A] [answered]
Hi, I want to set metric aggregation on field, if its total sum of values of the field is less than min_threshold then i should get an alert
but the sum is not working, every time its giving sum as zero even though in index field value is not zero, please help
```
name: low-packet
index: packet*

type: metric_aggregation

doc_type: doc

metric_agg_key: json.packetCount
metric_agg_type: sum
min_threshold: 10

alert:
 - slack

slack_webhook_url: "<slack_url>"
slack_username_override: "<name>"
slack_channel_override: "<slack_channel_name>"

alert_text: "COUNT OF PACKETS: '{0}'\n time: '{1}'\n"
alert_text_args: [metric_json.packetCount_sum, "@timestamp"]
```
I'm getting this alert
```
COUNT OF PACKETS: '0.0'
time: '2023-03-22T07:08:32.350660Z'
Threshold violation, sum:json.packetCount 0.0 (min: 10 max : None)
@timestamp: 2023-03-22T07:08:32.350660Z
metric_json.packetCount_sum: 0.0
num_hits: 0
num_matches: 1
```

**Comment by jertel** [comment, marked as answer]
In your `alert_text_args` parameter you specified `metric_json`, but in the `metric_agg_key` you specified `json`. Which one is the correct field name?

**Reply by the-arcade-01** [reply]
field name is `json.packetCount` and I specified `metric_json.packetCount_sum` in `alert_text_args` because as you can see in the output of the alert (below timestamp) we are getting 
``` 
metric_json.packetCount_sum: 0.0
num_hits: 0
num_matches: 1
```
so for displaying the text , i used it

**Reply by jertel** [reply]
Have you looked into the `buffer_time` window being used? Your rule above doesn't include that parameter so it will use the defaults.

I suggest enabling debug logging in ElastAlert 2. This will let you see the actual document field values, as well as the query being run against Elasticsearch.

**Reply by the-arcade-01** [reply]
yes @jertel, I can't enable debug logging, but after setting buffer_time, I'm getting the correct output
i want to ask, can i set `limit_execution` on this? and as per docs `limit_execution` only works in elastalert2,
is there any way to set it in previous version of elastalert?

**Reply by jertel** [reply]
The original version of ElastAlert has not been maintained for several years. This project/community is only involved with supporting ElastAlert 2.

I suggest prioritizing the upgrade to ElastAlert 2 over any new rule development.

**Reply by the-arcade-01** [reply]
okay, thanks @jertel

**Comment by the-arcade-01** [comment]
@jertel how do i specific a time range filter and will it interfere `buffer_time`? for the above metric_agg example, filter range should be 9am to 4pm
```
buffer_time:
 seconds: 900

filter:
- range:
    time:
      gte: "now/d+9h"
      lte: "now/d+16h"
```

**Reply by jertel** [reply]
ElastAlert 2 manages the time range. Attempting so specify your own time range filter will likely interfere and cause unexpected, unsupported results. There are several discussions that talk about different ways to influence the time span. Ex: #806. 

Separately, I've not attempted to specify math arguments (now/d + x) in the time parameters. That is something you will have to experiment with, as I don't know how ElastAlert 2 will handle it.

**Reply by the-arcade-01** [reply]
okay understood
